packages/react-ui-core/src/Gmap/Gmap.js

```diff
@@ -1,86 +1,162 @@
+
+@themed(/^Gmap/, { pure: true })
+
+export class Gmap extends PureComponent {
   static propTypes = {
-    apiKey: PropTypes.string.isRequired,
-    libraries: PropTypes.arrayOf(
-      PropTypes.string,
-    ),
-    spinner: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.node,
-      PropTypes.object,
-    ]),
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    children: PropTypes.node,
+    center: PropTypes.shape({
+      lat: PropTypes.number,
+      lng: PropTypes.number,
+    }),
+    zoom: PropTypes.number,
+    clickableIcons: PropTypes.bool,
+    mapControls: PropTypes.shape({
+      fullscreenControl: PropTypes.bool,
+      mapTypeControl: PropTypes.bool,
+      zoomControl: PropTypes.bool,
+      streetViewControl: PropTypes.bool,
+    }),
   }
 
   static defaultProps = {
-    libraries: [],
+    theme: {},
+    zoom: DEFAULT_ZOOM,
+    clickableIcons: false,
+    mapControls: {
+      fullscreenControl: false,
+      streetViewControl: false,
+    },
   }
 
   constructor(props) {
     super(props)
-    this.state = {
-      loaded: false,
-      spinner: props.spinner,
-    }
-  }
 
-  componentDidMount() {
-    if (this.isInitialized) {
-      this.scriptLoaded()
-      return
-    }
-    this.loadScript()
+    this.state = { map: null }
+    this.googleMap = React.createRef()
   }
 
-  get isInitialized() {
-    return !!(window.google && window.google.maps)
+  componentDidMount() {
+    this.initMap()
   }
 
-  get spinner() {
-    const { spinner, loaded } = this.state
-    const props = { key: 'map-spinner', loading: !loaded }
-
-    if (spinner) {
-      if (React.isValidElement(spinner)) return React.cloneElement(spinner, props)
-      return React.createElement(...parseArgs(spinner, 'div', props))
+  componentWillUnmount() {
+    if (this.map) {
+      window.google.maps.event.clearInstanceListeners(this.map)
     }
+  }
 
-    return null
+  setupEvents() {
+    EVENT_NAMES.forEach(name => {
+      if (this.props[name]) {
+        this.map.addListener(EVENTS[name], this.handleEvent(name))
+      }
+    })
   }
 
-  @autobind
-  scriptLoaded() {
-    this.setState({ loaded: true })
+  get children() {
+    const { map } = this.state
+    const { children } = this.props
+
+    return React.Children.map(React.Children.toArray(children), child =>
+      React.cloneElement(child, { map })
+    )
   }
 
-  async loadScript() {
-    const { apiKey, libraries } = this.props
-    const url = `${API_BASE_URL}?key=${apiKey}&libraries=${libraries.join()}`
+  initMap() {
+    const {
+      zoom,
+      mapControls,
+      center,
+      clickableIcons,
+    } = this.props
 
-    await import(/* webpackChunkName: "scriptjs" */ 'scriptjs').then(scriptjs => {
-      scriptjs(url, this.scriptLoaded)
+    this.map = new window.google.maps.Map(this.googleMap.current, {
+      zoom,
+      clickableIcons,
+      center: center || DEFAULT_CENTER,
+      ...mapControls,
     })
+
+    this.setupEvents()
```

Comment:
Does it make sense to call mapEventHelpers setupEvents() here since that logic exists?

Reply:
Nope, good catch. It should be using the helper for this.